Create str-join.md (#1520)
python-discord/bot: bot/resources/tags/str-join.md

**Joining Iterables**

> Suppose you want to nicely display a list (or some other iterable). The naive solution would be something like this.

Reviewer: ```suggestion
If you want to display a list (or some other iterable), you can write:
```

```py
colors = ['red', 'green', 'blue', 'yellow']
output = ""
separator = ", "
for color in colors:
    output += color + separator
print(output)
# Prints 'red, green, blue, yellow, '
```

> However, this solution is flawed. The separator is still added to the last element, and it is slow.

Reviewer: ```suggestion
However, the separator is still added to the last element, and it is relatively slow.
```

> The better solution is to use `str.join`.

Reviewer: ```suggestion
A better solution is to use `str.join`.
```

```py
colors = ['red', 'green', 'blue', 'yellow']
separator = ", "
print(separator.join(colors))
# Prints 'red, green, blue, yellow'
```

> This solution is much simpler, faster, and solves the problem of the extra separator. An important thing to note is that you can only `str.join` strings. For a list of ints,

Reviewer: ```suggestion
An important thing to note is that you can only `str.join` strings. For a list of ints,
```
